cmd/krel/cmd/ff.go

```diff
@@ -0,0 +1,307 @@
+		cmd.Stdin = strings.NewReader("some input")
+		var out bytes.Buffer
+		cmd.Stdout = &out
+		err = cmd.Run()
+		if err != nil {
+			log.Fatal(err)
+		}
+		log.Printf("in all caps: %q\n", out.String())
+		tags = append(tags, strings.TrimSuffix(out.String(), "\n"))
+	}
+
+	// TODO: This should return an error if it fails
+	// TODO: Provide more information to debug here
+	if tags[0] != tags[1] {
+		log.Printf("%s did not match %s", tags[0], tags[1])
+	}
+
+	// TODO: Rewrite using go-git
+	//       --dry-run appears to be unsupported for git push, so we shell out here.
+	checkPushCmd := exec.Command("git", "push", "-q", "--dry-run", util.KubernetesGitHubAuthURL)
+	util.Run(checkPushCmd)
+
+	w, err := repo.Worktree()
+	if err != nil {
+		return err
+	}
+
+	// ... checking out to commit
+	//Info("git checkout %s", commit)
+	remoteHash, err := repo.ResolveRevision(plumbing.Revision(fmt.Sprintf("%s/%s", remote, branch)))
+	if err != nil {
+		return err
+	}
+
+	err = w.Checkout(&git.CheckoutOptions{
+		Hash:   plumbing.NewHash(remoteHash.String()),
+		Branch: plumbing.NewBranchReferenceName(branch),
+		Create: true,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	// TODO: Merge and update
+	mergeCmd := exec.Command("git", "merge", "-X", "ours", remoteMaster)
+	util.Run(mergeCmd)
+
+	// TODO: Check for deleted files
+	// TODO: Do we need hack/install-etcd.sh
+	installEtcd := exec.Command("hack/install-etcd.sh")
+	util.Run(installEtcd)
+
+	currentPath := fmt.Sprintf("%s", os.Getenv("PATH"))
+	etcdPath := fmt.Sprintf("%s/third_party/etcd:%s", gitRoot, currentPath)
+	os.Setenv("PATH", os.Getenv("PATH"))
+	log.Printf("PATH has been set to %s", etcdPath)
+
+	// TODO: Running update scripts fails with go1.13
+	log.Printf("Running hack/update* scripts...")
+	updateScripts := []string{
+		"update-openapi-spec",
+	}
+	/*
+		for _, script := range updateScripts {
+			scriptPath := fmt.Sprintf("hack/%s.sh", script)
+			if _, err := os.Stat(scriptPath); os.IsNotExist(err) {
+				log.Printf("The update script (%s) does not exist. Skipping...", scriptPath)
+				continue
+			}
+
+			scriptCmd := exec.Command(scriptPath)
+			log.Printf("Running %s...", scriptPath)
+			util.Run(scriptCmd)
+		}
+	*/
+
+	status, err := w.Status()
+	if err != nil {
+		return err
+	}
+
+	if !status.IsClean() {
+		log.Printf("Commit changes:")
+		// TODO: Rewrite using go-git
+		gitAdd := exec.Command("git", "add", "-A")
+		util.Run(gitAdd)
+
+		// TODO: Rewrite using go-git
+		gitCommit := exec.Command("git", "commit", "-am", fmt.Sprintf("Results of running update scripts: %s", strings.Join(updateScripts, ",")))
+		util.Run(gitCommit)
+	}
+
+	releaseRefName := remoteHash.String()
+	releaseRev, err := util.RevParseShort(releaseRefName, gitRoot)
+	if err != nil {
+		return err
+	}
+
+	headRev, err := util.RevParseShort("HEAD", gitRoot)
+	if err != nil {
+		return err
+	}
+
+	log.Printf(
+		`Go look around in %s to make sure things look okay before pushing...
```

Comment:
nitpick: it might be more readable to move this to a separate variable outside the main function execution